src/com/swiper.js

```diff
@@ -0,0 +1,142 @@
+/**
+ * @author Deux Huit Huit
+ *
+ * Swiper component
+ *
+ * http://idangero.us/swiper/api
+ *
+ * @uses jQuery.fn.data()
+ * @requires https://cdnjs.cloudflare.com/ajax/libs/Swiper/4.5.1/js/swiper.min.js
+ */
+
+(function ($, win, undefined) {
+
+	'use strict';
+
+	App.components.exports('swiper', function (s) {
+		var scope = $();
+
+		var sels = $.extend({
+			element: '.js-swiper',
+			ctn: '.js-swiper-ctn',
+			wrapper: '.js-swiper-wrapper',
+			slide: '.js-swiper-slide',
+			scrollbar: '.js-swiper-scrollbar',
+			navigationPrev: '.js-swiper-navigation-previous',
+			navigationNext: '.js-swiper-navigation-next',
+			pagination: '.js-swiper-pagination',
+			preventClick: '.js-swiper-prevent-click'
+		}, s);
+
+		var nestedConfigs = [
+			'navigation',
+			'scrollbar',
+			'pagination',
+			'thumbs',
+			'autoplay',
+			'keyboard'
+		];
+
+		var defaultConfig = {
+			observeParents: true,
+			observer: true,
+			roundLengths: true
+		};
+
+		var DATA_KEY = 'swpr';
+
+		var swiper = function () {
+			return !!window.Swiper;
+		};
+
+		var filterByPrefix = function (obj, prefix) {
+			return _.pick(obj, function (value, key) {
+				return key.startsWith(prefix);
+			});
+		};
+
+		var clearPrefix = function (obj, prefix) {
+			var keys = Object.keys(obj);
+
+			keys.forEach(function (key) {
+				if (key.startsWith(prefix)) {
+					var newKey = key.substring(prefix.length);
+					newKey = newKey.charAt(0).toLowerCase() + newKey.slice(1);
+					obj[newKey] = obj[key];
+					delete obj[key];
+				}
+			});
+
+			return obj;
+		};
+
+		var getConfig = function (element, prefix) {
+			var config = clearPrefix(filterByPrefix(element.data(), prefix), prefix);
+
+			nestedConfigs.forEach(function (key) {
+				var data = filterByPrefix(config, key);
+				config = _.omit(config, _.keys(data));
+				var obj = clearPrefix(data, key);
+
+				if (_.size(obj)) {
+					config[key] = obj;
+				}
+			});
+
+			return config;
+		};
+
+		var init = function (element) {
```

Comment:
Our init methods always need a second params to add/override the default config of the swiper instence

Reply:
See flickity.js for reference